lib/phoenix/live_dashboard/components/table_component.ex

```diff
@@ -0,0 +1,213 @@
+defmodule Phoenix.LiveDashboard.TableComponent do
+  use Phoenix.LiveDashboard.Web, :live_component
+
+  @sort_dir ~w(desc asc)
+  @limit ~w(50 100 500 1000 5000)
+
+  @impl true
+  def mount(socket) do
+    {:ok, socket}
+  end
+
+  @impl true
+  def update(assigns, socket) do
+    %{
+      columns: columns,
+      row_fetcher: row_fetcher,
+      params: params,
+      self_path: self_path
+    } = assigns
+
+    limit_options = assigns[:limit_options] || @limit
+    columns = normalize_columns(columns)
+    params = normalize_table_params(params, columns, limit_options)
+    {rows, total} = row_fetcher.(params)
+
+    {:ok,
+     assign(socket,
+       columns: columns,
+       limit_options: limit_options,
+       list_name: assigns[:list_name] || "rows",
+       params: params,
+       row_attrs: assigns[:row_attrs] || [],
+       row_fetcher: row_fetcher,
+       rows: rows,
+       self_path: self_path,
+       title: assigns[:title] || "Table",
+       total: total || "unknown number"
```

Comment:
I believe we can raise if `title` or `total` are missing.

Reply:
For our current use cases calculate the total is not very expensive. I was thinking in showing rows in a database where calculate the total in some cases can be really expensive. Maybe a better solution could be just to show: "Showing at most 50 ports" and omitting the rest of the message.

Reply:
I think for now we can enforce it and if someone needs a different use case, they can open up an issue. :)

Reply:
Ok! I removed the "default value". In fact, we don't check if it is a number at all, so the developer can choose to return "unknown" or "unknown number" and it will work :)